src/util.c

```diff
@@ -334,3 +334,47 @@ get_global_network(int ip_family)
     else
         return "::/0";
 }
+
+struct netdef_pertype_iterator {
+    NetplanDefType type;
+    GHashTableIter iter;
+};
+
+NETPLAN_INTERNAL struct netdef_pertype_iterator*
+_netplan_iter_defs_per_devtype_init(const char *devtype)
+{
+    NetplanDefType type = netplan_def_type_from_name(devtype);
+    struct netdef_pertype_iterator *iter = g_malloc0(sizeof(*iter));
+    iter->type = type;
+    if (netdefs)
+        g_hash_table_iter_init(&iter->iter, netdefs);
+    return iter;
+}
+
+NETPLAN_INTERNAL NetplanNetDefinition*
+_netplan_iter_defs_per_devtype_next(struct netdef_pertype_iterator* it)
+{
+    gpointer key, value;
+
+    if (!netdefs)
+        return NULL;
+
+    while (g_hash_table_iter_next(&it->iter, &key, &value)) {
+        NetplanNetDefinition* netdef = value;
+        if (netdef->type == it->type)
+            return netdef;
+    }
+    return NULL;
+}
+
+NETPLAN_INTERNAL void
+_netplan_iter_defs_per_devtype_free(struct netdef_pertype_iterator* it)
+{
+    g_free(it);
+}
+
+NETPLAN_INTERNAL const char*
+_netplan_netdef_id(NetplanNetDefinition* nd)
```

Comment:
This is a rather generic function, that will probably be needed later on. Should we define it to be part of the `NETPLAN_PUBLIC` API and call it `netplan_netdef_get_id`?
We could still do this later on (once it's actually needed), as this is currently for internal use only.

Reply:
I indeed deliberately chose a symbol name that would be explicitly "unstable" (starting with `_`) because even though it is a generic function, I wasn't sure about the naming and didn't want to start shaving this particular yak in this PR ;).

When we settle on a proper scheme for the public accessors (I'm reserved about your proposal ;-) ) we can just alias this one or drop it completely if we decide on lockstep upgrade of `netplan.io` and `libnetplanX`.